CMakeLists.txt

```diff
@@ -0,0 +1,65 @@
+cmake_minimum_required(VERSION 3.0)
+
+# Compiler requirements
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+project(tinyply LANGUAGES CXX)
+set(PROJECT_VERSION 0.1)
+
+include_directories("${CMAKE_SOURCE_DIR}/source")
+
+# Library
+add_library(tinyply source/tinyply.cpp)
+target_include_directories(tinyply SYSTEM PUBLIC ${CMAKE_INSTALL_PREFIX}/include)
+
+# Example
+add_executable(example source/example.cpp)
+target_link_libraries(example PRIVATE tinyply)
```

Comment:
See [this excellent explanation](https://cmake.org/pipermail/cmake/2016-May/063400.html) on private linkage.  It doesn't surface here because `example.cpp` does `#include "tinyply.h` (as opposed to `#include <tinyply.h>`).  In short, `PRIVATE` vs `PUBLIC` here shouldn't matter at all if you get rid of `include_directories`, **and** everything remains in the same folder (which it should).  In my personal scrapped together CMake (where I don't care about installation, I build it from the parent project), I did something like

```cmake
# Build the example when no parent project
get_directory_property(TINYPLY_HAS_PARENT PARENT_DIRECTORY)
if (NOT TINY_PLY_HAS_PARENT)
  add_executable(tinyply-core "${CMAKE_CURRENT_SOURCE_DIR}/source/example.cpp")
  target_link_libraries(tinyply-core tinyply)
endif()
```

It may or may not be appropriate here since you aren't installing the executable (nor does it really make sense to).

Reply:
👍 Indeed private linkage here is irrelevant, I don't really know why I wrote that back then...